### ISSUE TITLE
Video in Office 2010 is possible after manual adjusting. Please make adjustmensts in code.

### ISSUE THREAD

**Issue by Laar2** [PHPOffice/PHPPresentation#491]
I am using PHPPresentation to create PPTX for Office 2010. When adding a video the PPTX must be repaired.
I compared some files and saw the following :

- in the PPTX this file is missing : \ppt\media\image1.png . This image is used as the black background in the video .
- in the ppt\slides\_rels\slide1.xml.rels the reference to this image1.png is also missing
- In the file ppt\slides\slide1.xml  the <a:blip r:embed="rId4"/> must reference this image1.png
- In the file ppt\slides\slide1.xml  the <**a:videoFile** r:link="rId2"/> must reference a video : <Relationship Id="rId2" Type="http://schemas.openxmlformats.org/officeDocument/2006/relationships/**video**" Target="../media/sintel_trailer-480p2.wmv"/>
- In the file ppt\slides\slide1.xml  the <p14:**media** xmlns:p14="http://schemas.microsoft.com/office/powerpoint/2010/main" r:embed="rId3"/> must referenece a media file :  <Relationship Id="rId3" Type="http://schemas.microsoft.com/office/2007/relationships/media" Target="../**media**/sintel_trailer-480p2.wmv"/>
- <cNvPr> has a duplicate ID, make these unique
- The entire <p:timing> bock is missing. the SPID values inside the timing block must match the second <cNvPr>

So when the missing image is there, the references to the media are all ok and the controles are linked (shape-target) to the right ID, then it works OK in Office 2010.

Is there someone still working on PHPpresentation?

**Comment by Laar2** [comment]
I started working on this...
I found a bug in AbstractSlide.php
line 1170 writing the media relation
`$objWriter->writeAttribute('r:embed', $shape->relationId);`

the same `$shape->relationId` is used when writing the video-relation

I added a function to add 1 to the relationId, which is a string;
`    /**
     * Increase the relationId.
     * @param string $currentRId
     * @return string new $currentRId
     */
     private function nextRId($currentRId) {
    	$prefix = "rId";
    	$nextnumberumber = substr($currentRId, strpos($currentRId, $prefix)+strlen($prefix));
    	$$nextnumberumber += 1;
    	return $prefix . $$nextnumberumber;
    }
`

so the new line will be :
`$objWriter->writeAttribute('r:embed', $this->nextRId($shape->relationId));`


To be continued.....

**Comment by Laar2** [comment]
It is working   :-)

I made several adjustments in the code and now i am able to add a video to the PPTX, without Powerpoint complaining about repairs.....

I reversed engineered the "Timing"- block in the Slide.
I also extended a Video-class from Media, just because I want to add audio to the PPTX using an Audio Class.
Which is working also.

Some fine-tuning is still needed.....

**Comment by Cryde** [comment]
Hello,

I'm facing an issue with video insertion (that need to be repair)

How did you solve this ? 
Is a patch available somewhere ?

Thank you

**Comment by Laar2** [comment]
I created my own version of PHPpresentation. I can send you the code.

It includes samples how to use the new classes Audio en Video.
But ONLY for PPTX, not for ODP!

**Comment by Laar2** [comment]
You can download the complete Eclipse project here :

https://laarnink.home.xs4all.nl/download/PhpOffice.zip

See the samples :
Sample_LA_audio.php
Sample_LA_video.php
Sample_LA_media.php

**Comment by gregsullivan** [comment]
@Laar2 Thanks for this—I'd begun noticing the same issues while manually comparing output for slides with video from PowerPoint with similar output from PHPPresentation, and found your fork just as I'd started doing the work of trying to resolve those mismatches myself.

Much appreciated! You say fine-tuning is needed; have you encountered any issues with PowerPoint compatibility so far?

**Comment by Laar2** [comment]
@gregsullivan I am using this version on one of my websites without any problems. 
I use:
- only the PPTX
- in MS Office 2010

I did not test or try other versions of Powerpoint.

**Comment by gregsullivan** [comment]
@Laar2 Thanks for letting me know, and thanks for making those changes!

To the extent it's useful for anyone else to know, this solved the problem for me on the latest Office 365 version of PowerPoint. Previously any presentations with videos would require repair, which removed the video.

**Comment by ashishnishad** [comment]
@Laar2 First of all thank you very much for providing audio support in ppt. But i have a query regarding auto play slides music in background so there is any method or function using it ppt slideshow automatically play music because currently i need to play it manually.
I want added audio automatically play when i start slideshow.
Thanks

**Comment by Laar2** [comment]
@ashishnishad 
I just had a quick look at it. 

Quick Workaround is maybe;
- create the pptx with the music you want to play on the first slide
- after storing the pptx file,  open it in Powerpoint and click the music icon in your first slide
- go to the playback tab and choose 'Automatically' in the start list
- save your powerpoint
- when playing the slideshow the music will start automatically.
See also : https://support.microsoft.com/en-gb/office/play-music-or-other-sounds-automatically-when-a-slide-appears-9a717c28-5365-4ee7-9515-c1d043bc7101


I'll try to find/build in this option in this version of PHPPresentation

**Comment by ashishnishad** [comment]
@Laar2 I have an issue with .wav extension audio, in PowerPoint showing codec unavailable.

**Comment by Laar2** [comment]
@ashishnishad 
Google is showing lots of solutions for that.
Have you tried mp3?

**Comment by ashishnishad** [comment]
@Laar2  Yeah, mp3 is working fine but i am recording voice from my website and then using in ppt and recorded audio in .wav format

**Comment by Laar2** [comment]
@ashishnishad
I don't think the problem is in PHPPresentation.
Maybe your recorded WAV does not have the right format/codec?
Maybe the codec you use on your website isn't present on the computer you play the wav (in PPTX) ??

Did you unzip the PPTX and tried to play the wav directly on your machine?

**Comment by ashishnishad** [comment]
@Laar2 Yes audio is playing fine in browser.

**Comment by ashishnishad** [comment]
@Laar2 Have you any idea about why exported ppt is not working in ppt 2007 version ?
